storm-core/src/jvm/org/apache/storm/utils/ConfigUtils.java

```diff
@@ -0,0 +1,711 @@
+        return (workerArtifactsRoot(conf, id) + FILE_SEPARATOR + port);
+    }
+
+    public static String workerArtifactsPidPath(Map conf, String id, Integer port) {
+        return (workerArtifactsRoot(conf, id, port) + FILE_SEPARATOR +  "worker.pid");
+    }
+
+    public static File getLogMetaDataFile(String fname) {
+        String[] subStrings = fname.split(FILE_SEPARATOR); // TODO: does this work well on windows?
+        String id = subStrings[0];
+        Integer port = Integer.parseInt(subStrings[1]);
+        return getLogMetaDataFile(Utils.readStormConfig(), id, port);
+    }
+
+    public static File getLogMetaDataFile(Map conf, String id, Integer port) {
+        String fname = workerArtifactsRoot(conf, id, port) + FILE_SEPARATOR + "worker.yaml";
+        return new File(fname);
+    }
+
+    public static File getWorkerDirFromRoot(String logRoot, String id, Integer port) {
+        return new File((logRoot + FILE_SEPARATOR + id + FILE_SEPARATOR + port));
+    }
+
+    public static String workerRoot(Map conf) {
+        return (absoluteStormLocalDir(conf) + FILE_SEPARATOR + "workers");
+    }
+
+    public static String workerRoot(Map conf, String id) {
+        return (workerRoot(conf) + FILE_SEPARATOR + id);
+    }
+
+    public static String workerPidsRoot(Map conf, String id) {
+        return (workerRoot(conf, id) + FILE_SEPARATOR + "pids");
+    }
+
+    public static String workerPidPath(Map conf, String id, String pid) {
+        return (workerPidsRoot(conf, id) + FILE_SEPARATOR + pid);
+    }
+
+    public static String workerHeartbeatsRoot(Map conf, String id) {
+        return (workerRoot(conf, id) + FILE_SEPARATOR + "heartbeats");
+    }
+
+    public static LocalState workerState(Map conf, String id) throws IOException {
+        return new LocalState(workerHeartbeatsRoot(conf, id));
+    }
+
+    public static Map overrideLoginConfigWithSystemProperty(Map conf) { // note that we delete the return value
+        String loginConfFile = System.getProperty("java.security.auth.login.config");
+        if (loginConfFile != null) {
+             conf.put("java.security.auth.login.config", loginConfFile);
+        }
+        return conf;
+    }
+
+    /* TODO: make sure test these two functions in manual tests */
+    public static List<String> getTopoLogsUsers(Map topologyConf) {
+        List<String> logsUsers = (List<String>)topologyConf.get(Config.LOGS_USERS);
+        List<String> topologyUsers = (List<String>)topologyConf.get(Config.TOPOLOGY_USERS);
+        Set<String> mergedUsers = new HashSet<String>();
+        if (logsUsers != null) {
+            for (String user : logsUsers) {
+                if (user != null) {
+                    mergedUsers.add(user);
+                }
+            }
+        }
+        if (topologyUsers != null) {
+            for (String user : topologyUsers) {
+                if (user != null) {
+                    mergedUsers.add(user);
+                }
+            }
+        }
```

Comment:
A reason for not doing so is that some user might be null or "null" in the configuration?
I see there is "remove nil?" in original clojure code.
``
  (sort (distinct (remove nil?
                    (concat
                      (topology-conf LOGS-GROUPS)
                      (topology-conf TOPOLOGY-GROUPS))))))
``